source/DSCResources/MSFT_SccmIniFIle/MSFT_SccmIniFIle.psm1

```diff
@@ -0,0 +1,1107 @@
+        $SQLDataFilePath,
+
+        [Parameter()]
+        [String]
+        $SQLLogFilePath,
+
+        [Parameter()]
+        [String]
+        $CCARSiteServer,
+
+        [Parameter()]
+        [String]
+        $CasRetryInterval,
+
+        [Parameter()]
+        [ValidateRange(0, 100)]
+        [uint16]
+        $WaitForCasTimeout,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $CloudConnector,
+
+        [Parameter()]
+        [String]
+        $CloudConnectorServer,
+
+        [Parameter()]
+        [Boolean]
+        $UseProxy,
+
+        [Parameter()]
+        [String]
+        $ProxyName,
+
+        [Parameter()]
+        [String]
+        $ProxyPort,
+
+        [Parameter()]
+        [Boolean]
+        $SAActive,
+
+        [Parameter()]
+        [Boolean]
+        $CurrentBranch
+    )
+
+    $IniFilePath = $IniFilePath.TrimEnd('\')
+    Write-Verbose "Getting file content of $IniFilePath\$IniFileName"
+    $iniContent = Get-Content -Path "$IniFilePath\$IniFileName" -ErrorAction SilentlyContinue
+    $result = $true
+
+    if ($iniContent)
+    {
+        foreach ($line in $iniContent)
+        {
+            if ($line -match '=')
+            {
+                $iniParameters += @{
+                    (($line.split('=')[0]).Trim(' ')) = (($line.split('=')[1]).Trim(' '))
+                }
+            }
+        }
+
+        $systemParameters = @('Verbose','Debug','ErrorAction','WarningAction','InformationAction','ErrorVariable','WarningVariable','InformationVariable','OutVariable','OutBuffer','PipelineVariable')
+        $PSBoundParameters.Remove('IniFilePath') | Out-Null
+        $PSBoundParameters.Remove('IniFileName') | Out-Null
+
+        foreach ($param in $PSBoundParameters.GetEnumerator())
+        {
+            switch ($param.Value)
+            {
+                $true   {$newValue = 1}
+                $false  {$newValue = 0}
+                default {$newValue = $param.Value}
+            }
+            if ($iniParameters.$($param.Key) -and $iniParameters.$($param.Key) -ne $newValue)
+            {
+                Write-Verbose -Message "NOTMATCH: $($param.Key) - Current Value: $($iniParameters.$($param.Key)) Target Value: $newValue"
+                $result = $false
+            }
+            elseif (-not $iniParameters.$($param.Key) -and $systemParameters -notcontains $param.Key)
+            {
+                Write-Verbose -Message "NOTMATCH: $($param.Key) - Current Value: `$null Target Value: $newValue"
+                $result = $false
+            }
+            elseif ($iniParameters.$($param.Key) -and $iniParameters.$($param.Key) -eq $newValue)
+            {
+                Write-Verbose -Message "Match: $($param.Key) - Current Value: $($iniParameters.$($param.Key)) Target Value: $newValue"
+            }
+        }
+
+    }
+    else {
+        $result = $false
+    }
+
+    Write-Verbose -Message "Test returned: $result."
+    return $result
+} #end function Test-TargetResource
```

Comment:
need to end on an empty line and also should also include:

Export-ModuleMember -Function *-TargetResource

Reply:
Fixed.